tests/linalg_test.py

```diff
@@ -2114,6 +2122,58 @@ def testToeplitzConstructionWithKnownCases(self):
       [2, 1, 4, 5],
       [3, 2, 1, 4]], dtype=np.float32))
 
+  @jtu.sample_product(
+    shape=[(2, 3), (4, 6), (5, 7), (100, 300)],
+    dtype = float_types + complex_types
+  )
+  def test_solve_sylvester(self, shape, dtype):
```

Comment:
This test is failing on TPU because non-symmetric `eig` is not implemented. You can fix this by specifying that the test should only run on CPU and GPU; see the `eig` tests for an example of this: https://github.com/jax-ml/jax/blob/da106b971af143906c8cbec8f4f2dfb7b57a11ce/tests/linalg_test.py#L268-L269